server/src/main/java/org/apache/iotdb/db/utils/FilePathUtils.java

```diff
@@ -195,43 +171,49 @@ public static String getTsFilePrefixPath(TsFileResource resource) {
       if (newRecord.getFields().get(i) == null) {
         aggregateResultList.add(
             AggregateResultFactory.getAggrResultByName(
-                plan.getAggregations().get(i), plan.getDeduplicatedDataTypes().get(i)));
+                plan.getDeduplicatedAggregations().get(i), plan.getDeduplicatedDataTypes().get(i)));
       } else {
         TSDataType dataType = newRecord.getFields().get(i).getDataType();
         AggregateResult aggRet =
-            AggregateResultFactory.getAggrResultByName(plan.getAggregations().get(i), dataType);
-        switch (dataType) {
-          case TEXT:
-            aggRet.setBinaryValue(newRecord.getFields().get(i).getBinaryV());
-            break;
-          case INT32:
-            aggRet.setIntValue(newRecord.getFields().get(i).getIntV());
-            break;
-          case INT64:
-            aggRet.setLongValue(newRecord.getFields().get(i).getLongV());
-            break;
-          case FLOAT:
-            aggRet.setFloatValue(newRecord.getFields().get(i).getFloatV());
-            break;
-          case DOUBLE:
-            aggRet.setDoubleValue(newRecord.getFields().get(i).getDoubleV());
-            break;
-          case BOOLEAN:
-            aggRet.setBooleanValue(newRecord.getFields().get(i).getBoolV());
-            break;
-          default:
-            throw new UnSupportedDataTypeException(dataType.toString());
+            AggregateResultFactory.getAggrResultByName(
+                plan.getDeduplicatedAggregations().get(i), dataType);
+        if (aggRet.getAggregationType().equals(AggregationType.AVG)) {
+          ((AvgAggrResult) aggRet).updateAvg(dataType, newRecord.getFields().get(i).getDoubleV());
+        } else {
```

Comment:
If it's just for here, I think it's better to write a new public method `setAvg`. which set `avg` and `cnt` directly rather than modify the access restrictions of `private` method.